tests/backends/test_kafka.py

```diff
@@ -231,6 +231,90 @@ def test_consumer_stream_processor_shutdown(self) -> None:
                 with pytest.raises(RuntimeError):
                     processor.run()
 
+    @mock.patch("arroyo.processing.processor.BACKPRESSURE_THRESHOLD", 0)
+    def test_assign_partition_during_pause(self) -> None:
```

Comment:
Can we run the test twice. One with `handle_poll_while_paused` set to True and one set to False so we can see that one fails and the other does not ?